libs/proposals/src/lib/proposals-data-provider/proposals-data-provider.tsx

```diff
@@ -104,7 +115,25 @@ export const update: Update<
 
 const getMarketProposalsData = (
   responseData: MarketViewProposalsQuery | null
-) => removePaginationWrapper(responseData?.proposalsConnection?.edges) || [];
+): MarketViewProposalFieldsFragment[] => {
+  const proposals: MarketViewProposalFieldsFragment[] = [];
+  responseData?.proposalsConnection?.edges?.forEach((edge) => {
+    if (edge?.proposalNode) {
+      if (edge.proposalNode.__typename === 'Proposal') {
+        proposals.push(edge.proposalNode as MarketViewProposalFieldsFragment);
+      } else if (
+        edge.proposalNode.__typename === 'BatchProposal' &&
+        edge.proposalNode.subProposals
+      ) {
+        proposals.push(
+          ...(edge.proposalNode
+            .subProposals as MarketViewProposalFieldsFragment[])
```

Comment:
Rather than type casting here, which could be problematic why not defined the type of the data provider correctly? Right now when used in a component you will expect to only receive MarketViewProposalFieldsFragment, which is not correct, as you might actually have a sub proposal.

What I would suggest is to adjust the fragments slight, and then chagne the type of the data provider and helper functions. Something like this:

```
fragment SubProposal on ProposalDetail {
  ...
}

fragment MarketViewProposalFields {
  ...
}

query MarketViewProposals(
  $proposalType: ProposalType
  $inState: ProposalState
) {
  proposalsConnection(proposalType: $proposalType, inState: $inState) {
    edges {
      proposalNode {
        ... on BatchProposal {
          id
          subProposals {
             ...SubProposal
          }
        }
        ... on Proposal {
          ...MarketViewProposalFields
        }
      }
    }
  }
}

```

Then in data dataprovider we would need to update the types to reflect a combined array of 'normal' proposals and sub proposals
```
type Proposals = Array<MarketViewProposalFieldsFragment | SubProposalFieldsFragment>;

export const marketViewProposalsDataProvider = makeDataProvider<
  MarketViewProposalsQuery,
  Proposals,
  MarketViewLiveProposalsSubscription,
  MarketViewProposalFieldsFragment,
  MarketViewProposalsQueryVariables,
  MarketViewLiveProposalsSubscriptionVariables
>({
  query: MarketViewProposalsDocument,
  subscriptionQuery: MarketViewLiveProposalsDocument,
  update,
  getDelta: (subscriptionData: MarketViewLiveProposalsSubscription) =>
    subscriptionData.proposals,
  getData: getMarketProposalsData,
  getSubscriptionVariables: () => subscriptionVariables,
});

```

Reply:
I have updated